composer/cli/launcher.py

```diff
@@ -197,8 +197,13 @@ def _parse_args():
     if args.nproc < 1:
         raise ValueError('The nproc must be 1 or greater')
 
-    if args.world_size is None and 'WORLD_SIZE' in os.environ:
-        args.world_size = int(os.environ['WORLD_SIZE'])
+    if args.world_size is None:
+        if 'WORLD_SIZE' in os.environ and os.environ.get('LOCAL_WORLD_SIZE') != os.environ['WORLD_SIZE']:
```

Comment:
tiniest of nits: should store this boolean expression as a variable: `using_multinode`. Only would be useful if there was more logic for multinode vs. not

Reply:
It's a little tricky since we define multinode variable later on. Technicallyyy you can define WOLRD_SIZE and then manually set `LOCAL_WORLD_SIZE` via nproc, so this wouldn't always work.